### DISCUSSION TITLE
CSLA 5.5.0 EF Core Calling SQL Proc with Parameters

### DISCUSSION THREAD

**Discussion by jhuerta3** [MarimerLLC/csla#2467] [Questions]
Hello All,
Im working on a Entity Frame Core CSLA 5.5 prototype project which I am almost done. I haven't done EF in a long time; Its been ADO for the most part but I have done CSLA for a few years now. The only thing left is to get a SQL procedure call working. The issue is that when I pass values from the SqlParameter list any which are null/empty have the value of "default" E.g (@employeeid=default,)  on the SQL server. Im not sure if Im doing this correctly. I was thinking maybe there is another class that I should be calling to convert the values into correct DBNull values when passed back to the server. If I change all the "default" values to "null" on the script that is sent to the SQL server it works. Below is code that I am using in the DataAccess.Dal class. By the way if I modify the proc to only take two parameters I and pass values for the two it works fine. Its only when they are empty. 
Thank you. 

        public List<FarInquiryEntity> Get(List<SqlParameter> parms)
        {
            string sql = "EXEC usp_farInquiry " +
                           "@ain,@employeeid,@runno,@RunNoTo,@rundate" +
                           ",@RunDateTo,@batchno,@BatchNoTo,@DocumentNo" +
                           ",@pdbCluster,@pdbDivisionIndex,@TXNCode" +
                           ",@region,@status,@FARCode,@postkey" +
                           ",@RespUnit,@TXNYear,@PageSize,@PageNumber" +
                           ",@SortCol,@SortDir";

            var data = _context.FarInquirys.FromSqlRaw<FarInquiryEntity>(sql, parms.ToArray()).ToList();
            if (data != null)
                return data;
            else
                throw new KeyNotFoundException($"Far Inquiry Not Found");
        }

**Comment by rockfordlhotka** [comment]
I always look for content from Julie Lerman (The Data Farm) when I have EF questions - which is common, as I don't use EF all that often myself 😏

**Reply by jhuerta3** [reply]
Will do. T
Thank you Sir and have a good day!
